clang/lib/AST/Type.cpp

```diff
@@ -3666,11 +3673,14 @@ void FunctionProtoType::Profile(llvm::FoldingSetNodeID &ID, QualType Result,
   // Finally we have a trailing return type flag (bool)
   // combined with AArch64 SME Attributes, to save space:
   //      int
+  // Then add the FunctionEffects
   //
   // There is no ambiguity between the consumed arguments and an empty EH
   // spec because of the leading 'bool' which unambiguously indicates
   // whether the following bool is the EH spec or part of the arguments.
 
+  ID.AddPointer(epi.FunctionEffects.getOpaqueValue()); // TODO: Where???
```

Comment:
Thanks; this call seems to be taking a lot of care about this but it does seem to relate to how various parts are optional.

Reply:
Yeah, you’re putting the effects first, which seems fine so long as you always add the effects (you’re right in that there seem to be some optional fields, so I’d be wary about moving it further down); updating the comment to reflect where the effects are encoded is a good idea—I didn’t notice that at first because the diff didn’t include that part of the comment.